src/vuex/modules/distribution.js

```diff
@@ -88,30 +87,31 @@ export default ({ node }) => {
         commit(`setDistributionError`, error)
       }
     },
-    async simulateWithdrawAllRewards({ rootState: { session }, dispatch }) {
+    async simulateWithdralRewards({ rootState: { session }, dispatch }) {
       return await dispatch(`simulateTx`, {
         type: `MsgWithdrawDelegationReward`,
         txArguments: {
-          toAddress: session.address,
-          validatorAddresses: []
+          toAddress: session.address
         }
       })
     },
-    async withdrawAllRewards(
+    async withdrawRewards(
       { rootState, getters, dispatch },
-      { gas, gasPrice, denom, validatorAddress, password, submitType }
+      { gas, gasPrice, denom, password, submitType }
     ) {
-      const top5Delegations = getTop5Delegations(getters.committedDelegations)
+      // Compares the amount in a [address1, {denom: amount}] array
+      const byDenom = denom => (a, b) => b[1][denom] - a[1][denom]
 
-      const validatorAddresses = validatorAddress
-        ? [validatorAddress]
-        : Object.keys(top5Delegations)
+      const validatorList = Object.entries(getters.distribution.rewards || [])
```

Comment:
```suggestion
      const validatorList = Object.entries(getters.distribution.rewards) || []
```

Reply:
In case rewards was undefined. Now that you mention it, it was a little premature. Rewards defaults to {}, and Object.entries will return an empty array on it's own.